### ISSUE TITLE
Images do not load on the ListView in iPhone 5 but works on iPhone 6 and up

### ISSUE THREAD

**Issue by arunshan** [facebook/react-native#1345]
In ListView, the images do not load properly on the iPhone 5/5s but loads on the iPhone 6. Both the devices are on iOS 8.3. 

iPhone 5/5s: 

![img_2543](https://cloud.githubusercontent.com/assets/1124102/7710453/cf00c194-fe18-11e4-8f74-d6afbbf9ed55.PNG)

iPhone 6:

![img_0145](https://cloud.githubusercontent.com/assets/1124102/7710494/29d2a16e-fe19-11e4-832a-5fbfb1f07e50.PNG)

In the following code I replaced the authorization headers with "**********************". 

```javascript
'use strict';

var React = require('react-native');
var Dimensions = require('Dimensions');
var ExampleWebViewStandalone = require('./ExampleWebView');
var RefreshableListView = require('react-native-refreshable-listview');
var Viewport = require('react-native-viewport');

var {
  Image,
  ListView,
  TouchableHighlight,
  StyleSheet,
  Text,
  Component,
  View,
  AppRegistry,
  NavigatorIOS,
  LinkingIOS
} = React;

var experience_url = 'http://showcase.example.com/api/experiences';
var showcase_url = 'http://showcase.example.com/api/showcases';

var ExampleListView = React.createClass({

  getInitialState: function() {
    if(this.props.navigatorobj)
      this.props.navigatorobj.setState({
        navBarHidden : false
      });

    var ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 !== r2});
    this._fetchData();
    return {
      dataStore : ds,
      dataSource : ds.cloneWithRows([]),
      loaded: false
    };
  },

  componentWillMount: function() {
    var that = this;
    LinkingIOS.addEventListener('url', function(event){
      that.props.slugname = event.url.split("/").pop();
      that._fetchData();
    });
  },

  componentDidUnmount: function() {
  },

  componentDidMount: function() {
    var that = this;
    that._fetchData();
    LinkingIOS.addEventListener('url', function(event){
      that.props.slugname = event.url.split("/").pop();
      that._fetchData();
    });
  },

  _fetchData: function() {
    var self = this;
    var projectids = [];
    var urls = [];
    var properties = self.props;
    fetch(showcase_url,{
      headers: { authorization: "Basic **********************" }
    })
    .then((response) => response.json())
    .then((responseData) => {
      responseData.map(function(showcase, i){
        var slugname = properties.slugname ? properties.slugname : 'exampleapp';
        if(showcase.slug === slugname) {
          projectids = showcase.project_ids;
        }
      });
      fetch(experience_url,{
        headers: { authorization: "Basic **********************" }
      })
      .then((response) => response.json())
      .then((responseData) => {
        projectids.map(function(projectid,i){
          responseData.map(function(experience, i){
            if(experience.project_id === projectid){
              urls.push({
                'bgurl' : experience.showcase.background_image_url,
                'title' : experience.showcase.title,
                'uri':'http://experiences.example.com/'+experience.release._id+"/index.html"
              });
            }
          });
        });
        self.setState({
          dataSource: self.state.dataStore.cloneWithRows(urls),
          loaded: true
        });
      })
      .done();
    })
    .done();
  },

  _pressRow: function(uri) {
    this.props.navigator.push({
      component: ExampleWebViewStandalone,
      passProps:{
        url: uri
      },
      onLeftButtonPress: () => {
        console.log("The left button is pressed");
      }
    });
  },

  _renderRow: function(rowData) {
    console.log("the url is ", rowData.bgurl);
    return (
      <TouchableHighlight onPress={() => this._pressRow(rowData.uri)}>
        <View>
          <View style={styles.row}>
            <Image style={styles.thumb}
              source={{uri : rowData.bgurl}}
              defaultSource={{uri: 'http://ktteradio.com/wp-content/themes/LiveTheme_223/images/video-placeholder.jpg'}} >
            <Text style={styles.text}>
              {rowData.title}
            </Text>
            </Image>
          </View>
          <View style={styles.separator} />
        </View>
      </TouchableHighlight>
    );
  },

  goBack: function() {
  },

  render: function() {
    return (
        <RefreshableListView
          dataSource={this.state.dataSource}
          renderRow={(rowData) => this._renderRow(rowData)}
          loadData={this._fetchData}
        />
    );
  },
});

var styles = StyleSheet.create({
  container: {
    flex: 1,
    width: Dimensions.get("window").width,
    height: Dimensions.get("window").height
  },

  row: {
    flexDirection: 'row',
    justifyContent: 'center',
    backgroundColor: '#F6F6F6',
  },

  separator: {
    height: 0,
    backgroundColor: '#CCCCCC',
  },

  thumb: {
    width: Dimensions.get("window").width,
    height: Dimensions.get("window").height-85,
    backgroundColor: 'rgba(0,0,0,0)',
    justifyContent: 'center',
    alignItems: 'center'
  },

  text: {
    fontSize: 20,
    textAlign: 'center',
    backgroundColor: 'rgba(0,0,0,0)',
    color: 'white',
    fontWeight: 'bold',
    fontFamily: 'Helvetica',
  }
});

module.exports = ExampleListView;

```

**Comment by rxb** [comment]
Might be the same issue, I am also having trouble loading some, but not all images in a ListView on a 5s. Works perfectly in the simulator.

**Comment by arunshan** [comment]
@rxb do you know what may be the reason ? Are there any image size or image format restrictions that I am not aware of ? List view render is invoked only after the image urls are received from the server. I believe there is some kind of race condition.

**Comment by rxb** [comment]
@arunshan Still trying to parse out what's different about the ones that are intermittently not loading.

**Comment by unknownexception** [comment]
Would you mind to try https://github.com/facebook/react-native/pull/1318 and see that `onError` handler returns? It could help to make `Image` error handler better.

**Comment by arunshan** [comment]
@unknownexception It works now! Thanks.

**Comment by christopherdro** [comment]
@arunshan Would you mind closing this if it's been solved?

**Comment by arunshan** [comment]
I just noticed something really strange. For some reason when the app is downloaded for the first time, then few images do not load. But works fine when I close the app and reopen it on the iPhone. It happens on the iPhone 6 with iOS 8.3. It is hard to reproduce the error. I need to do a hard reset on the simulator and check if the onerror prop is being invoked again. Will give you an update later today.

**Comment by brentvatne** [comment]
@arunshan - if that issue persists, please open a new issue that describes it in detail and your best efforts at reproducing it :smile: - thanks for reporting!
